proctor-common/src/main/java/com/indeed/proctor/common/ForceGroupsOptionsStrings.java

```diff
@@ -17,60 +26,161 @@
  *     <li>It's a concatenation of string elements separated by commas. (e.g., my_btn_tst1,default_to_fallback)</li>
  *     <li>Each element represents a forced test group, or an option.</li>
  *     <li>A forced group is specified by a test name followed by a bucket value (e.g., my_btn_tst1)</li>
+ *     <li>A forced payload is specified by a force group followed by a semicolon and a payload definition (e.g., my_btn_tst1;doubleValue:0.2) </li>
  *     <li>A option is specified by predefined tokens that doesn't contain integers (e.g., default_to_fallback) </li>
  *     <li>If two elements conflict (e.g., specifying different buckets for the same test),
  *     the latter takes precedence</li>
  * </ul>
  */
 public class ForceGroupsOptionsStrings {
     private static final Logger LOGGER = LogManager.getLogger(ForceGroupsOptionsStrings.class);
+    private static final int FORCE_PARAMETER_BUCKET_IDX = 0;
+    private static final int FORCE_PARAMETER_PAYLOAD_IDX = 1;
+    private static final int FORCE_PARAMETER_MAX_SIZE = 2;
 
     private ForceGroupsOptionsStrings() {
     }
 
     @Nonnull
-    public static ForceGroupsOptions parseForceGroupsString(@Nullable final String forceGroupsString) {
+    public static ForceGroupsOptions parseForceGroupsString(@Nullable final String forceGroupsString, final Set<String> forcePayloadTests) {
         final ForceGroupsOptions.Builder builder = ForceGroupsOptions.builder();
         if (forceGroupsString == null) {
             return builder.build();
         }
-        // using single char in split regex avoids Pattern creation since java8
-        final String[] pieces = forceGroupsString.split(",");
-        // detect integer number from end of string
-        for (final String rawPiece : pieces) {
-            final String piece = rawPiece.trim();
-            if (piece.isEmpty()) {
+
+        // \w+-?\d+;\w+:(?s)(\".*?\"(?<!\\\")|\[.+?\](?!\"|\])|-?\d+\.?\d*) - matches force payload
+        //      \w+-?\d+;\w+: - matches force group followed by ; and payload type (ie. example_tst1;stringValue)
+        //                    followed by one of the following:
+        //          (\".*?\"(?<!\\\") - matches a string ignoring escaped quotes
+        //          (\[.+?\](?!\"|\]) - matches an array with cases where string array may contain brackets within the string
+        //          (-?\d+\.?\d*)       - matches a integer or double
+        // \w+-?\d+ - matches force group without payload (ie. example_tst1)
+        // [a-z_]+ - matches default value (ie. default_to_min_live)
+        final Pattern pattern = Pattern.compile("\\w+-?\\d+;\\w+:(\\\".*?\\\"(?<!\\\\\\\")|\\[.+?\\](?!\\\"|\\])|-?\\d+\\.?\\d*)|\\w+-?\\d+|[a-z_]+");
```

Comment:
nitpick: it's nice to define this variable as `static final` so that we can avoid compiling the same regular expression every time when the method is called.